homeassistant/components/ecobee/climate.py

```diff
@@ -31,13 +32,14 @@
 PRESET_HOLD_INDEFINITE = 'indefinite'
 AWAY_MODE = 'awayMode'
 
-ECOBEE_HVAC_TO_HASS = {
-    'auxHeatOnly': HVAC_MODE_HEAT,
-    'heat': HVAC_MODE_HEAT,
-    'cool': HVAC_MODE_COOL,
-    'off': HVAC_MODE_OFF,
-    'auto': HVAC_MODE_AUTO,
-}
+# Order matters, because for reverse mapping we don't want to map HEAT to AUX
+ECOBEE_HVAC_TO_HASS = collections.OrderedDict([
+    ('heat', HVAC_MODE_HEAT),
+    ('cool', HVAC_MODE_COOL),
+    ('auto', HVAC_MODE_AUTO),
```

Comment:
I know this wasn't the scope of this PR, but this should be mapped to `HVAC_MODE_HEAT_COOL`.  Ecobee allows you to set heat/cool setpoints when in "Auto" mode.  This is true for probably nearly all of the US thermostats and anyone using "auto" in the new schema likely mapped incorrectly.

Reply:
https://github.com/home-assistant/architecture/issues/260

Reply:
Yeah, hence why I opened that issue -- here is an example of where it was done incorrectly under the new schema. :)

Reply:
should not block this pull thou since it fixes broken test in master